src/pages/iou/IOUDetailsModal.js

```diff
@@ -77,29 +81,109 @@ class IOUDetailsModal extends Component {
+        if (props.iouReport.submitterPayPalMeAddress) {
+            paymentOptions.push(CONST.IOU.PAYMENT_TYPE.PAYPAL_ME);
+        }
+
+        this.submitterPhoneNumber = undefined;
+        this.isComponentMounted = false;
+
         this.state = {
-            // Temporarily placeholder, as the Venmo/Paypal issue will introduce settlement types.
-            settlementType: 'Elsewhere',
+            paymentType: CONST.IOU.PAYMENT_TYPE.ELSEWHERE,
+            isSettlementMenuVisible: false,
+            paymentOptions,
         };
 
         this.performIOUPayment = this.performIOUPayment.bind(this);
     }
 
     componentDidMount() {
+        this.isComponentMounted = true;
         fetchIOUReportByID(this.props.route.params.iouReportID, this.props.route.params.chatReportID);
+        this.addVenmoPaymentOptionIfAvailable();
+    }
+
+    componentWillUnmount() {
+        this.isComponentMounted = false;
+    }
+
+    setMenuVisibility(isSettlementMenuVisible) {
+        this.setState({isSettlementMenuVisible});
     }
 
     performIOUPayment() {
         payIOUReport({
             chatReportID: this.props.route.params.chatReportID,
             reportID: this.props.route.params.iouReportID,
-            paymentMethodType: this.state.settlementType,
+            paymentMethodType: this.state.paymentType,
+            amount: this.props.iouReport.total,
+            currency: this.props.iouReport.currency,
+            submitterPayPalMeAddress: this.props.iouReport.submitterPayPalMeAddress,
+            submitterPhoneNumber: this.submitterPhoneNumber,
         });
     }
 
+    /**
+     * @param {String} phoneNumber
+     * @returns {Boolean}
+     */
+    isValidUSPhone(phoneNumber) {
+        // Remove alphanumeric characters and validate that this is in fact a phone number
+        return CONST.REGEX.PHONE_E164_PLUS.test(phoneNumber.replace(CONST.REGEX.NON_ALPHA_NUMERIC, ''))
+
+            // Next make sure it's a US phone number
+            && CONST.REGEX.US_PHONE.test(phoneNumber);
+    }
+
+    /**
+     * Checks to see if we can use Venmo. The following conditions must be met:
+     *
+     *   1. The IOU report currency is USD
+     *   2. The submitter has as a valid US phone number
+     *   3. Venmo app is installed
+     *
+     */
+    addVenmoPaymentOptionIfAvailable() {
+        if (this.props.iouReport.currency !== CONST.CURRENCY.USD) {
+            return;
+        }
+
+        const submitterPhoneNumbers = lodashGet(this.props.iouReport, 'submitterPhoneNumbers', []);
+
+        if (_.isEmpty(submitterPhoneNumbers)) {
+            return;
+        }
+
+        this.submitterPhoneNumber = _.find(submitterPhoneNumbers, this.isValidUSPhone);
+
+        if (!this.submitterPhoneNumber) {
+            return;
+        }
+
+        isAppInstalled('venmo')
+            .then((isVenmoInstalled) => {
+                if (!isVenmoInstalled || !this.isComponentMounted) {
+                    return;
+                }
```

Comment:
NAB: Maybe add a note explaining why `isComponentMounted` is a requirement? 

Is it because isAppInstalled is Async, and the user might have navigated away and unmounted the Comp?